REFERENCE: QGA-6

KristineZelenivska opened the pull request:
DESCRIPTION: Made authentication middleware. also created file
   for authController and auth Route

Commit:
REFERENCE: QGA-6
   DESCRIPTION: Made authentication middleware. also created file
   for authController and auth Route

Gruseckis commented on server.js:
```diff
@@ -26,6 +26,8 @@ app.use(bodyParser.urlencoded({ extended: true }));
 
 // Routing
 
+
```
Remove white spaces

Gruseckis commented on middlewares/authenticate.js:
```diff
@@ -0,0 +1,39 @@
+    });
+
+const authenticate = async (req, res, next) => {
+    if (process.env.SKIP_AUTH) {
```
Remove this since we are not using this

Gruseckis commented on middlewares/authenticate.js:
```diff
@@ -0,0 +1,39 @@
+
+        if (decodedToken && decodedToken.data && decodedToken.data.username) {
+            const { username } = decodedToken.data;
+            const user = await getUserByName(username);
```
getUserByName is undefined

Commit:
REFERENCE: QGA-6
   DESCRIPTION: Deleted white spaces, removed SKIP_AUTH part and
   defined getUserByName

Commit:
REFERENCE: QGA-6
   DESCRIPTION: restored userModel file to resolve conflict,
   changed import of only function i use, also changed function name.

Gruseckis commented on .env.template:
```diff
@@ -1,11 +0,0 @@
-MONGODB_URI='mongodb://localhost:27017/quiz-game'
```
Please restore this file

Commit:
REFERENCE: QGA-6
  DESCRIPTION: restored .env.template file

Gruseckis commented on middlewares/authenticate.js:
```diff
@@ -0,0 +1,39 @@
+            const { username } = decodedToken.data;
+            const user = await getUserByUsername(username);
+            if (user) {
+                logger.log('debug', `User: ${username} was successfully authenticated`);
```
Change level from 'debug' to 'info'

Commit:
REFERENCE: QGA-6
  DESCRIPTION: change logger format

Commit:
REFERENCE: QGA-6

Commit:
REFERENCE: QGA-6

Commit:
REFERENCE: QGA-6
   DESCRIPTION: made git fetch and got all development branch in.
